codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -332,7 +332,29 @@ public static void toHttp2Headers(HttpHeaders inHeaders, Http2Headers out) throw
                 // https://tools.ietf.org/html/rfc7540#section-8.1.2.2 makes a special exception for TE
                 if (!aName.contentEqualsIgnoreCase(HttpHeaderNames.TE) ||
                     AsciiString.contentEqualsIgnoreCase(entry.getValue(), HttpHeaderValues.TRAILERS)) {
-                    out.add(aName, AsciiString.of(entry.getValue()));
+                    if (aName.contentEqualsIgnoreCase(HttpHeaderNames.COOKIE)) {
```

Comment:
https://tools.ietf.org/html/rfc7540#section-8.1.2.2
> The only exception to this is the TE header field, which MAY be
   present in an HTTP/2 request; when it is, it MUST NOT contain any
   value other than "trailers".